### DISCUSSION TITLE
How to use rich with Python curses?

### DISCUSSION THREAD

**Discussion by dev-SR** [Textualize/rich#2252] [Q&A]
I'm thinking of creating a CLI app using python `curses` and `rich` . As `curses` requires `addstr` method to print text to string, I'm unable to print using `rich`. Is it possible to integrate these two libraries?

Following codes don't work accordingly!!

```python
import curses
from curses import wrapper
from rich.console import Console
console = Console()
with console.capture() as capture:
    console.print("[bold red]Hello[/] World")
str_output = capture.get()

def main(stdscr):
    stdscr.clear()
    stdscr.addstr(str_output)
    stdscr.refresh()
    stdscr.getch()

wrapper(main)
```

**Comment by USLTD** [comment]
@dev-SR, Unfortunately that's how most of **curses** implementations, including **ncurses** are. They just don't recognize ANSI sequences as whole so they just split them resulting it for fail (I mean that ANSI sequence fails to get interpreted by terminal).
Thankfully there is at least 1 project for Python which solves this problem: https://github.com/getcuia/cusser
It solves problem by wrapping **stdscr** and interpreting ANSI sequences.
I haven't tested it yet but it should work.
